ksqldb-common/src/main/java/io/confluent/ksql/function/AggregateFunctionInitArguments.java

```diff
@@ -28,18 +30,41 @@
  */
 public class AggregateFunctionInitArguments {
 
+  public static final AggregateFunctionInitArguments EMPTY_ARGS =
+      new AggregateFunctionInitArguments();
+
   private final int udafIndex;
   private final List<Object> initArgs;
+  private final Map<String, ?> config;
 
-  public static final AggregateFunctionInitArguments EMPTY_ARGS =
-      new AggregateFunctionInitArguments();
+  /**
+   * This method should only be used for legacy "built-in" UDAF
+   * implementations that implement AggregateFunctionFactory directly
+   * such as TopKAggregateFuncitonFactory. Otherwise, the config will
+   * not be properly passed through to the aggregate function.
+   */
+  public AggregateFunctionInitArguments(
+      final int index,
+      final Object... initArgs
+  ) {
+    this(index, ImmutableMap.of(/* not a configurable function */), Arrays.asList(initArgs));
+  }
 
-  public AggregateFunctionInitArguments(final int index, final Object... initArgs) {
-    this(index, Arrays.asList(initArgs));
+  public AggregateFunctionInitArguments(
+      final int index,
+      final Map<String, ?> config,
+      final Object... initArgs
+  ) {
+    this(index, config, Arrays.asList(initArgs));
   }
 
-  public AggregateFunctionInitArguments(final int index, final List<Object> initArgs) {
+  public AggregateFunctionInitArguments(
+      final int index,
+      final Map<String, ?> config,
+      final List<Object> initArgs
+  ) {
     this.udafIndex = index;
+    this.config = Objects.requireNonNull(config, "config");
```

Comment:
Should we call `ImmutableMap.copyOf(config)` so we keep the config completely immutable?